salt/metalk8s/repo/redhat.sls

```diff
@@ -41,29 +60,25 @@ Ensure yum versionlock plugin is enabled:
       - pkg: Install yum-plugin-versionlock
 
 {%- for repo_name, repo_config in repo.repositories.items() %}
-  {%- if repo.local_mode %}
-    {%- set repo_base_url = "file://" ~
-                            salt.metalk8s.get_archives()[saltenv].path ~ "/" ~
-                            repo.relative_path ~ "/" ~
-                            grains['os_family'].lower() ~ "/" ~
-                            "$metalk8s_osmajorrelease" %}
-  {%- else %}
-    {%- set repo_base_url = "http://" ~ repo_host ~ ':' ~ repo_port ~
-                            "/" ~ saltenv ~ "/" ~
-                            grains['os_family'].lower() ~ "/" ~
-                            "$metalk8s_osmajorrelease" %}
-  {%- endif %}
-  {%- set repo_url = repo_base_url ~ "/" ~ repo_name ~
-                     "-el$metalk8s_osmajorrelease" %}
+  {%- set repo_urls = [] %}
+  {%- for base_url in repo_base_urls %}
+    {%- do repo_urls.append(base_url ~ "/" ~
+                            repo_name ~ "-el$metalk8s_osmajorrelease") %}
+  {%- endfor %}
   {%- set gpg_keys = [] %}
   {%- for gpgkey in repo_config.gpgkeys %}
-    {%- do gpg_keys.append(repo_url ~ "/" ~ gpgkey) %}
+    {# NOTE: We only use the first repo url as gpg key since the keys are the same,
+             and we do not want to depends on both repos to import this key,
+       When installing a package and the gpg key of the package is not already imported
+       on the host, it will try to import **all** gpgkeys listed and fail if we are
+       unable to download one of them #}
```

Comment:
```suggestion
    {# NOTE: We only use the first repo URL for GPG key since these are the same,
       and we do not want to depend on both repos to import this key.
       When installing a package, if the package's GPG key is not already imported
       on the host, it will try to import **all** listed GPG keys and will fail if unable to
       download one of them #}
```

Reply:
Well, but it means it the Bootstrap node is down, we can't install new packages ?
Can't we just pre isntall the GPG keys from all repositories on all nodes ?

Reply:
I'm not sure that's an issue for now, because packages are only needed when deploying stuff and the salt-master is not HA yet, so if the bootstrap node is down... we can't deploy anything anyway.

Reply:
If the gpg key is already imported it's ok even if the "gpgkey" listed is not reachable. 
So to me, we could keep the bootstrap node only for the moment since we need bootstrap to deploy the node anyway, it means we will import those keys when deploying the node anyway.

But I see what you mean we could manually import those keys first and then we are good, yes sure the only "issue" is that I need to manually do all the "yum variable" replacements in Salt... and not sure it's worth doing it

Reply:
Right, just if you "upgrade" from rhel7 to rhel8 you can keep repo definition but ... not sure we care

Reply:
Yes, but if you upgrade, you will likely completely reinstall the node and you still need to change the variable value (it's easier, I agree, but not sure it worth it).
I think if someone upgrade, he will have some other troubles and will probably have to re-run an highstate anyway, fixing this at the time.

Reply:
Yes totally agree

Anyway, I will do this in this PR :smile: